redis/src/test/java/org/apache/calcite/adapter/redis/RedisAdapterCaseBase.java

```diff
@@ -0,0 +1,176 @@
+ */
+public class RedisAdapterCaseBase extends RedisDataCaseBase {
+  /**
+   * URL of the "redis-zips" model.
+   */
+  private URL url =
+      RedisAdapterCaseBase.class.getResource("/redis-mix-model.json");
+
+  private String targetFile;
+  private boolean isSupport = true;
+
+  @SuppressWarnings("unchecked")
+  private static final Map<String, Integer> TABLE_MAPS = new HashMap(15);
+
+  static {
+    TABLE_MAPS.put("raw_01", 1);
+    TABLE_MAPS.put("raw_02", 2);
+    TABLE_MAPS.put("raw_03", 2);
+    TABLE_MAPS.put("raw_04", 2);
+    TABLE_MAPS.put("raw_05", 2);
+    TABLE_MAPS.put("csv_01", 1);
+    TABLE_MAPS.put("csv_02", 2);
+    TABLE_MAPS.put("csv_03", 2);
+    TABLE_MAPS.put("csv_04", 2);
+    TABLE_MAPS.put("csv_05", 2);
+    TABLE_MAPS.put("json_01", 1);
+    TABLE_MAPS.put("json_02", 2);
+    TABLE_MAPS.put("json_03", 2);
+    TABLE_MAPS.put("json_04", 2);
+    TABLE_MAPS.put("json_05", 2);
+  }
+
+  @BeforeEach
+  @Override public void makeData() {
+    super.makeData();
+    setRedisJsonFile();
+  }
+
+  /**
+   * Whether to run this test.
+   */
+  private boolean enabled() {
+    return CalciteSystemProperty.TEST_REDIS.value();
+  }
+
+  /**
+   * Creates a query against a data set given by a map.
+   */
+  private CalciteAssert.AssertQuery sql(String sql, URL url) {
+    return CalciteAssert.that()
+        .enable(enabled())
+        .withModel(url)
+        .query(sql);
+  }
+
+  /**
+   * Creates a query against the {@link #url} data set.
+   */
+  private CalciteAssert.AssertQuery sql(String sql) {
+    return sql(sql, url);
+  }
+
+  private void setRedisJsonFile() {
+    try {
+      String strResult;
+      ObjectMapper objMapper = new ObjectMapper();
+      objMapper.configure(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES, true)
+          .configure(JsonParser.Feature.ALLOW_SINGLE_QUOTES, true)
+          .configure(JsonParser.Feature.ALLOW_COMMENTS, true);
+      File file = new File(this.url.getFile());
+      if (file.exists()) {
+        JsonNode rootNode = objMapper.readTree(new File(this.url.getFile()));
+        strResult = rootNode.toString().replace(Integer.toString(Protocol.DEFAULT_PORT),
+            Integer.toString(PORT));
+        String tmpPath = System.getProperty("java.io.tmpdir");
+        targetFile = tmpPath + "/" + System.currentTimeMillis() + ".json";
+        try (OutputStream fos = new FileOutputStream(targetFile, false);
+             Writer writer = new OutputStreamWriter(fos, StandardCharsets.UTF_8)) {
+          writer.write(strResult);
+          writer.flush();
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
+        url = new File(targetFile).toURI().toURL();
+      } else {
+        isSupport = false;
+      }
+    } catch (Exception e) {
+      isSupport = false;
+      e.printStackTrace();
+    }
+  }
+
+  @Test
+  public void testRedisBySql() {
+    if (!isSupport) {
+      return;
+    }
```

Comment:
What does that mean? Does that mean the test would silently be marked as "passed test" while in practice it would be ignored/skipped?

Please write tests in such a way so "successful" test in logs means the test was really performing some verifications.

Reply:
> What does that mean? Does that mean the test would silently be marked as "passed test" while in practice it would be ignored/skipped?
> 
> Please write tests in such a way so "successful" test in logs means the test was really performing some verifications.

This process is because some CI found a problem reading and writing the file some time ago, now I have modified this logic.